File: materials/2019-2020/11v/2019-12-18-polymorphism-exercise/circle.cc
#include "circle.hh"
#include <iostream>
#include <cmath>

Circle::Circle(Point const& center, double radius)
  : center(center), radius(radius) {}

void Circle::draw() const {
  std::cout
    << "Circle:\n"
    << "\tCenter: "
    << center
    << '\n'
    << "\tRadius: "
    << radius;
}

double Circle::get_perimeter() const {
  return 2 * M_PI * radius;
}

double Circle::get_area() const {
  return M_PI * radius * radius;
}

File: materials/2019-2020/11a/practice/2020-02-26-unit-tests/1va_zad/src/main/java/com/elsys/Circle.java
package main.java.com.elsys;

import static java.lang.StrictMath.pow;

public class Circle {
    Point O;
    double r;

    Circle(Point O, int r){
        this.O=O;
        this.r=r;
    }
    public double getPerimeter(){
        return (double) (6.28*r);
    }
    public double getArea(){
        return (double) ((double) 3.14*pow(r, 2));
    }
}
